test/SILOptimizer/array_element_propagation.sil

```diff
@@ -222,7 +222,6 @@ sil @unknown_use : $@convention(thin) () -> () {
 // CHECK:      [[ASFUN:%.*]] = function_ref @arrayAdoptStorage
 // CHECK-NEXT: [[ARR:%.*]] = apply [[ASFUN]]
 // CHECK-NEXT: [[OWNER:%.*]] = tuple_extract [[ARR]]{{.*}}, 0
-// CHECK-NOT:    apply [[ACFUN]]
```

Comment:
The `CHECK-NOT` lines are actually essential for all the tests. They check if the optimization could really remove those instructions. I know, they didn't have an effect so far, but it should be easy to fix those, e.g. by removing the argument lit-variable. E.g.
```
// CHECK-NOT:    apply
```

Reply:
There are other applies between the preceding CHECK-NEXT and the following CHECK, so the test fails.
Specifically the application of the arrayInit and the array meta type.

Reply:
The current output between the arrayAdoptStorage and arrayAppendContentsOf is
```sil
  // function_ref arrayAdoptStorage
  %5 = function_ref @arrayAdoptStorage : $@convention(thin) (@owned AnyObject, MyInt, @thin Array<MyInt>.Type) -> @owned (Array<MyInt>, UnsafeMutablePointer<MyInt>) // user: %6
  %6 = apply %5(%3, %2, %4) : $@convention(thin) (@owned AnyObject, MyInt, @thin Array<MyInt>.Type) -> @owned (Array<MyInt>, UnsafeMutablePointer<MyInt>) // users: %8, %7
  %7 = tuple_extract %6 : $(Array<MyInt>, UnsafeMutablePointer<MyInt>), 0 // user: %25
  %8 = tuple_extract %6 : $(Array<MyInt>, UnsafeMutablePointer<MyInt>), 1 // user: %9
  %9 = struct_extract %8 : $UnsafeMutablePointer<MyInt>, #UnsafeMutablePointer._rawValue // user: %10
  %10 = pointer_to_address %9 : $Builtin.RawPointer to [strict] $*MyInt // user: %13
  %11 = integer_literal $Builtin.Int64, 27        // user: %12
  %12 = struct $MyInt (%11 : $Builtin.Int64)      // users: %22, %13
  store %12 to %10 : $*MyInt                      // id: %13
  %14 = alloc_stack $Array<MyInt>                 // users: %23, %26, %18
  %15 = metatype $@thin Array<MyInt>.Type         // user: %17
  // function_ref arrayInit
  %16 = function_ref @arrayInit : $@convention(method) (@thin Array<MyInt>.Type) -> @owned Array<MyInt> // user: %17
  %17 = apply %16(%15) : $@convention(method) (@thin Array<MyInt>.Type) -> @owned Array<MyInt> // user: %18
  store %17 to %14 : $*Array<MyInt>               // id: %18
  // function_ref arrayAppendContentsOf
  %19 = function_ref @arrayAppendContentsOf : $@convention(method) (@owned Array<MyInt>, @inout Array<MyInt>) -> ()
  // function_ref Array.append(_:)
  %20 = function_ref @$sSa6appendyyxnF : $@convention(method) <τ_0_0> (@in τ_0_0, @inout Array<τ_0_0>) -> () // user: %23
```
We are specifically getting hung up on the application of `arrayInit`.

Reply:
ok, then there should be a
```
[[ACFUNC:%.*]] = function_ref @arrayAppendContentsOf
```
before the `CHECK-NOT: apply [[ACFUNC]]`.
The point here is that this function is not called.

Reply:
Done